### PR TITLE
qt: Handle fonts of deleted widgets properly, streamline the flow in `GUIUtil::updateFonts`

### LINKED ISSUE #3767: [qt] Abnormal non-deterministic scaling of Settings text
This was happening on 34a146c5c8e5e5242c92b297f70ac9fbe82fc842, a commit in https://github.com/dashpay/dash/pull/3682, however I have no reason to believe that it is related to that PR.

See the video showing these weird scaling stuff. Happening on Ubuntu 18.04
https://youtu.be/_i7BNwukRU4

### PR DESCRIPTION
**Pull request by UdjinM6**
Fixes #3767

### CODE REVIEW

**Comment by xdustinface** [comment]
@UdjinM6 I can confirm this solves the issues pointed out in #3767 (FYI, i was able to reproduce it on macOS also).

But this fix introduces a new scaling bug due to font inheritance (Just start the wallet with a very low/high font scale to reproduce it in most screens).

See [suggestion-3772](https://github.com/xdustinface/dash/tree/suggestion-3772) which contains a fix   (8c075b98bc9ef4587777d2b9dc86c2460dc4dad0) for the introduced issue and also some more refactoring in `GUIUtil::updateFonts`.

**Comment by UdjinM6** [comment]
@xdustinface Good catch! Cherry-picked them all 👍

**Review by xdustinface — APPROVED** [review]
ACK

File: src/qt/guiutil.cpp [hunk 1]
```diff
@@ -1556,86 +1583,59 @@ void updateFonts()
         if (std::find(vecIgnore.begin(), vecIgnore.end(), w->metaObject()->className()) != vecIgnore.end()) {
             continue;
         }
+        ++nUpdatable;
+
         QFont font = w->font();
+        assert(font.pointSize() > 0);
```
Comment (PastaPastaPasta) [inline src/qt/guiutil.cpp:1589]: I really don't love using asserts... If they do catch they provide basically no info to the user for us to debug with...

Comment (UdjinM6) [reply src/qt/guiutil.cpp:1589]: `>0` means that it's either a default font size (set via `setApplicationFont`) or some another valid value and font size was _not_ set via `setPixelSize`. IMO it's safe to use it here, should only crash if we break smth in code (e.g. use `setPixelSize`).

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**